### PR TITLE
[Practice] Reverting changes

### PR DESCRIPTION
**Pull request by peternied**
### Description
The maintainers had a discussion on how reverts have been done in the
past and how we would like to ensure the codebase is always ready for
contribution.  The theme of that discussion was prioritizing
stable and available codebase - documenting that in this change.

### Check List
- [ ] ~New functionality includes testing~
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cliu123** [comment]
Are we make reverting commits a part of regular basis practices? Based on my understanding of OpenSearch best practices, reverting should be minimized. But this is a good topic to discuss on organization level. @peternied Thanks for bringing this up! Thoughts? @opensearch-project/opensearch-core (I don't see a tag for all OpenSearch maintainers.)

File: MAINTAINERS.md
Comment (peternied) [inline MAINTAINERS.md:23]: > Are we make reverting commits a part of regular basis practices?

@cliu123 Could you include the language that you'd like to see changed or your recommendation you have?

File: MAINTAINERS.md
Comment (peternied) [inline MAINTAINERS.md:4]: > But this is a good topic to discuss on organization level.

@cliu123 Different repositories inside OpenSearch-Project will have different requirements to avoid all of those situations, I am tailoring this document around our repo first.

**Comment by andrross** [comment]
> Based on my understanding of OpenSearch best practices, reverting should be minimized.

We should certainly strive towards not introducing commits that cause impact, and we should continuously review and update processes that give us confidence that commits are good before merging them. However, no process is perfect and we're all human, so mistakes will be made. If a commit is merged and causes problems, then the revert-first-then-investigate approach is very reasonable because it is the fastest way to ensure that other folks do not remain blocked by the problem. Reverts are a fast and safe way to get a code base back to a known-good state (and are easy to undo themselves) so I don't see any particular reason to be wary of them. I'd love to hear if anyone has a differing opinion.

**Comment by cliu123** [comment]
> > Based on my understanding of OpenSearch best practices, reverting should be minimized.
> 
> We should certainly strive towards not introducing commits that cause impact, and we should continuously review and update processes that give us confidence that commits are good before merging them. However, no process is perfect and we're all human, so mistakes will be made. If a commit is merged and causes problems, then the revert-first-then-investigate approach is very reasonable because it is the fastest way to ensure that other folks do not remain blocked by the problem. Reverts are a fast and safe way to get a code base back to a known-good state (and are easy to undo themselves) so I don't see any particular reason to be wary of them. I'd love to hear if anyone has a differing opinion.

Totally agree. The PR itself brings up a good topic, so no wary at all. My take is that the entire organization should have a consistent process/policy/documentation on the process of reverting commit.

File: MAINTAINERS.md
Comment (cliu123) [reply MAINTAINERS.md:23]: The language looks good to me. My question is on the process of reverting and how we can minimize revert.

File: MAINTAINERS.md
Comment (owaiskazi19) [inline MAINTAINERS.md:26]: This is a great addition. A small question on the same line. Are we having a firm timeline to resolve the issue in an hour or should we be more specific by providing an hour as an example?

File: MAINTAINERS.md
Comment (owaiskazi19) [inline MAINTAINERS.md:31]:
```suggestion
Reverts can also be done via the command line using `git revert <commit_id>` and creating a new pull request.  If done in this 
```

Comment (peternied) [reply MAINTAINERS.md]: Done in [1ba40dd](https://github.com/opensearch-project/security/pull/1754/commits/1ba40dd5e7aa2a8516e65e207f44f0501cbc536e)

File: MAINTAINERS.md
Comment (peternied) [reply MAINTAINERS.md]: This is documented firm to ensure issues are resolved quickly.  If we see a reason to modify this amount of time this document can be updated.

File: MAINTAINERS.md
Comment (davidlago) [reply MAINTAINERS.md:23]: If the to-be-reverted change is blocking, failing builds or disrupting in other ways, and a fix forward is not quick, we should revert and work on a fix that reverts the revert plus any other changes needed to fix the initial issue. @cliu123 I believe that's what you are pointing at, as in reverting should be a last resort to unblock rather than the go-to strategy for fixing things, and I think that's exactly what that first paragraph on this PR conveys... but if it's not clear please feel free to suggest re-wording.

**Review by davidlago — APPROVED** [review]

**Review by cliu123 — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]

**Comment by owaiskazi19** [comment]
> @andrross @owaiskazi19 Thanks again from the thoughtful comments - consider bringing this topic to your own teams or adding practices like this in the [.github](https://github.com/opensearch-project/.github) repo.

Sure. Can you please create an issue on OpenSearch? We can take it from there.

**Comment by peternied** [comment]
FYI - Manually backport associated with #2843

**Pull request merged**
